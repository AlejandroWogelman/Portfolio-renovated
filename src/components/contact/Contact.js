import React from "react";
import "./contact.css";

export const Contact = () => {
  return (
    <section className="contact-container">
      <h2 className="contact-title" id="contact">
        contact
      </h2>
      <div className="contact-mail">
        <a href="mailto:alehellamn@gmail.com">alehellamn@gmail.com</a>
      </div>
      <ul className="ul-nav">
        <li>
          <a
            href="https://github.com/AlejandroWogelman?tab=repositories"
            target="_blank"
            rel="noreferrer"
            className="contact-icon"
          >
            <i className="fab fa-github"></i>
          </a>
        </li>
        <li>
          <a
            href="https://www.linkedin.com/in/alejandrowogel?lipi=urn%3Ali%3Apage%3Ad_flagship3_profile_view_base_contact_details%3B%2BvtBYjh%2FQrCJ4tOULZ7mag%3D%3D"
            target="_blank"
            rel="noreferrer"
            className="contact-icon"
          >
            <i className="fab fa-linkedin-in"></i>
          </a>
        </li>
        <li>
          <a
            href="https://www.facebook.com/AlejandroWogel/"
            target="_blank"
            rel="noreferrer"
            className="contact-icon"
          >
            <i className="fab fa-facebook-f"></i>
          </a>
        </li>
      </ul>
    </section>
  );
};
