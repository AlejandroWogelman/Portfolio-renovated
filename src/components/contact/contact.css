.contact-container {
  display: flex;
  flex-direction: column;
  min-height: 300px;
  background-image: url(../../assets/chat-svgrepo-com.svg);

  background-repeat: no-repeat;
  background-size: 20%;
  background-color: rgb(30, 43, 57);
  background-position: 105% 25%;
}
.contact-title {
  font-family: "Comfortaa", cursive;
  font-size: 2.3rem;
  text-align: center;
  color: rgb(233, 233, 233);
  padding: 3rem 0;
}
.contact-mail {
  text-align: center;
  padding-bottom: 1rem;
}
.contact-mail a {
  transition: color 0.3s;
  margin: 0 auto;
  font-family: "Comfortaa", cursive;
  color: rgb(255, 255, 255);
  font-size: 1.3rem;
}
.contact-mail :hover {
  color: rgba(255, 255, 255, 0.52);
}
@media screen and (max-width: 720px) {
  .contact-container {
    margin-bottom: 4.2rem;
  }
}
.ul-nav {
  display: flex;
  padding: 1rem;
  justify-content: center;
}

.contact-icon {
  font-size: 3rem;
  color: rgb(0, 0, 0);
}
.contact-icon i {
  transition: all 0.3s;
}
.contact-icon i:hover {
  transform: rotate(-15deg);
}
