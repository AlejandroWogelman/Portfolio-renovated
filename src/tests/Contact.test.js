import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { Contact } from "../components/contact/Contact";

describe("<Contact/>", () => {
  const wrapper = shallow(<Contact />);
  test("Contact rendering", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("URL from contacts is correct", () => {
    const anchor = wrapper.find("li");
    const Git = anchor.at(0).find("a").prop("href");
    const Linkedin = anchor.at(1).find("a").prop("href");
    const Facebook = anchor.at(2).find("a").prop("href");

    expect(Git).toBe("https://github.com/AlejandroWogelman?tab=repositories");

    expect(Linkedin).toBe(
      "https://www.linkedin.com/in/alejandrowogel?lipi=urn%3Ali%3Apage%3Ad_flagship3_profile_view_base_contact_details%3B%2BvtBYjh%2FQrCJ4tOULZ7mag%3D%3D"
    );

    expect(Facebook).toBe("https://www.facebook.com/AlejandroWogel/");
  });
});
