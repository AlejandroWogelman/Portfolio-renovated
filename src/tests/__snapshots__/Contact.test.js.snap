// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact/> Contact rendering 1`] = `
<section
  className="contact-container"
>
  <h2
    className="contact-title"
    id="contact"
  >
    contact
  </h2>
  <div
    className="contact-mail"
  >
    <a
      href="mailto:alehellamn@gmail.com"
    >
      alehellamn@gmail.com
    </a>
  </div>
  <ul
    className="ul-nav"
  >
    <li>
      <a
        className="contact-icon"
        href="https://github.com/AlejandroWogelman?tab=repositories"
        rel="noreferrer"
        target="_blank"
      >
        <i
          className="fab fa-github"
        />
      </a>
    </li>
    <li>
      <a
        className="contact-icon"
        href="https://www.linkedin.com/in/alejandrowogel?lipi=urn%3Ali%3Apage%3Ad_flagship3_profile_view_base_contact_details%3B%2BvtBYjh%2FQrCJ4tOULZ7mag%3D%3D"
        rel="noreferrer"
        target="_blank"
      >
        <i
          className="fab fa-linkedin-in"
        />
      </a>
    </li>
    <li>
      <a
        className="contact-icon"
        href="https://www.facebook.com/AlejandroWogel/"
        rel="noreferrer"
        target="_blank"
      >
        <i
          className="fab fa-facebook-f"
        />
      </a>
    </li>
  </ul>
</section>
`;
